libs/hdf-converters/src/msft-secure-score-mapper.ts

```diff
@@ -0,0 +1,203 @@
+                }
+              },
+              userImpacts: {
+                transformer: (data: ControlScore) => {
+                  // return userImpacts from the profile document where its id matches the controlName
+                  return this.getProfiles(data.controlName || '')
+                    .filter((profile) => profile.userImpact !== undefined)
+                    .map((profile) => profile.userImpact);
+                }
+              },
+              nist: ['SA-11', 'RA-5']
+            },
+            source_location: {},
+            descriptions: [
+              {
+                data: {
+                  transformer: (
+                    data: ControlScore & {implementationStatus: string}
+                  ) => {
+                    const profiles = this.getProfiles(data.controlName || '');
+                    const remediationSteps = profiles
+                      .map((profile: SecureScoreControlProfile) =>
+                        profile.remediation?.toString()
+                      )
+                      .filter(
+                        (remediation: string | undefined) =>
+                          remediation !== undefined
+                      );
+
+                    return remediationSteps.join('\n');
+                  }
+                },
+                label: 'fix'
+              }
+            ],
+            results: [
+              {
+                status: {
+                  transformer: (
+                    data: ControlScore & {scoreInPercentage: number}
+                  ) => {
+                    if (data.scoreInPercentage === 100) {
+                      return ExecJSON.ControlResultStatus.Passed;
+                    }
+
+                    const knownMaxScores = this.getProfiles(
+                      data.controlName || ''
+                    ).map((profile) => profile.maxScore || 0);
+
+                    const highMaxScore = Math.max(...knownMaxScores);
+
+                    if (knownMaxScores.length === 0) {
+                      // no Profile found matching the controlName
+                      return ExecJSON.ControlResultStatus.Failed;
+                    } else if (data.score === undefined) {
+                      return ExecJSON.ControlResultStatus.Error;
+                    } else if (data.score === highMaxScore) {
+                      return ExecJSON.ControlResultStatus.Passed;
+                    } else {
+                      return ExecJSON.ControlResultStatus.Failed;
+                    }
+                  }
+                },
+                code_desc: {
+                  transformer: (
+                    data: ControlScore & {implementationStatus: string}
+                  ) => data.implementationStatus
+                },
+                start_time: {transformer: () => this.data.createdDateTime}
+              }
+            ]
+          }
+        ],
+        sha256: ''
+      }
+    ],
+    passthrough: {
```

Comment:
a key thing that we try to do during the mapping process is ensure that all data is brought through into the hdf file, even if it just gets dumped into passthrough.

we are missing significant sections of the output at the moment.  i've already mentioned the URLs.  we also have secureScore.averageComparitiveScores, securescore.enabledServices, securescore.vendorInformation, etc.  we still have a couple fields in controlscore that are unmapped such as lastsynced.  doing the wholesale dump of profiles is fine.